src/main/resources/com/puppycrawl/tools/checkstyle/grammar/java.g

```diff
@@ -758,6 +758,7 @@ annotationDefault
 enumBlock
     :    LCURLY
             ( enumConstant ( options{greedy=true;}: COMMA enumConstant )* ( COMMA )? )?
+            ( COMMA )?
```

Comment:
Basically this is:
````
( ( A B A )* B? )?
B?
````

I don't understand why a second `B?` is needed when it seems already covered.

Does it may any sense to make this:
````
( ( A B A )* )?
B?
````
maybe even:
````
( A B A )*
B?
````

This way there is only one `B?` or does it have the same issue?

Reply:
Yes, this expression looks like it can be refined to:

```
( A ( B A )* )?
B?
```
This is consistent with the [JLS](https://docs.oracle.com/javase/specs/jls/se15/html/jls-8.html#jls-8.9) grammar:

_EnumBody_:
{ _[_EnumConstantList_] [,] [_EnumBodyDeclarations_]_ }
_EnumConstantList_:
_EnumConstant {, EnumConstant}_
...

Reply:
Good catch, fixed